ports/zephyr/mpconfigport.h

```diff
@@ -113,6 +118,12 @@
 #define MICROPY_COMP_CONST (0)
 #define MICROPY_COMP_DOUBLE_TUPLE_ASSIGN (0)
 
+#ifdef CONFIG_THREAD_CUSTOM_DATA
```

Comment:
It's a bit unconventional that a Zephyr option controls a MicroPython option.  But I can't really see a better way to do it.

Maybe add a comment here along the lines of "When the Zephyr CONFIG_THREAD_CUSTOM_DATA feature is enabled, the MicroPython MICROPY_PY_THREAD feature is enabled automatically"